pkg/processor/build/builder.go

```diff
@@ -1626,3 +1617,18 @@ func (b *Builder) getFunctionTempFile(tempDir string, functionPath string, isArc
 	// for non-archives, must retain file name
 	return os.OpenFile(path.Join(tempDir, functionPathBase), os.O_RDWR|os.O_CREATE, 0600)
 }
+
+func (b *Builder) getBuildTimeoutSeconds() int64 {
+	if b.options.FunctionConfig.Spec.Build.BuildTimeoutSeconds != nil {
+		if *b.options.FunctionConfig.Spec.Build.BuildTimeoutSeconds > 0 {
+			return *b.options.FunctionConfig.Spec.Build.BuildTimeoutSeconds
+		}
+
+		// no timeout
+		return math.MaxInt64 - time.Now().UnixNano()
```

Comment:
Previous behavior also took negative (like `-1`) to be "no timeout". please don't change that